src/main/java/org/fundacionjala/virtualassistant/service/SpeechService.java

```diff
@@ -0,0 +1,39 @@
+package org.fundacionjala.virtualassistant.service;
+
+import org.fundacionjala.virtualassistant.repository.ASRClient;
+import org.fundacionjala.virtualassistant.repository.WhisperClient;
+import org.springframework.stereotype.Service;
+import java.io.IOException;
+import java.nio.file.Path;
+
+@Service
+public class SpeechService {
+
+    private ASRClient asrClient;
+
+    public SpeechService(ASRClient asrClient) {
+        this.asrClient = asrClient;
+    }
+
+    public SpeechService() {
+        try {
+            asrClient = new WhisperClient();
```

Comment:
here you are adding a dependency to the WhisperClient, please define a field in the class to refer to the ASRClient and that should be incharged of calling whisper